auth-foundation/src/main/java/com/okta/authfoundation/credential/Credential.kt

```diff
@@ -106,6 +105,85 @@ class Credential internal constructor(
         }
     }
 
+    companion object {
+        private val defaultCredentialIdDataStore: DefaultCredentialIdDataStore
+            get() = DefaultCredentialIdDataStore.instance
+
+        @VisibleForTesting
+        internal suspend fun credentialDataSource() = CredentialDataSource.getInstance()
+
+        /**
+         * Returns the default [Credential], or null if no default [Credential] exists.
+         */
+        suspend fun getDefaultCredential(): Credential? {
```

Comment:
Is it common practice in Kotlin/Java to have different getter/setter functions? Or is it common practice to create a var with custom getter/setter functions?

I'm wondering because it feels like it would be more convenient to be able to say `Credential.default` instead of `Credential.getDefaultCredential()`

Reply:
This stems from a limitation. Default credential is fetched using coroutine APIs, but properties can't access coroutine APIs without using runBlocking.